Fireblast/streamlit_app.py

```diff
@@ -0,0 +1,732 @@
+        selected_query_files = []
+        defined_queries = []
+        queries_dir = "queries"
+        sql_files = []
+        if os.path.isdir(queries_dir):
+            sql_files = [f for f in os.listdir(queries_dir) if f.endswith(".sql")]
+        if use_defined_queries:
+            if sql_files:
+                for sql_file in sql_files:
+                    checked = st.checkbox(sql_file, value=False, key=f"qps_query_{sql_file}")
+                    if checked:
+                        selected_query_files.append(sql_file)
+                # Load the SQL from selected files
+                for fname in selected_query_files:
+                    with open(os.path.join(queries_dir, fname), "r") as f:
+                        defined_queries.append(f.read())
+            else:
+                st.warning("No 'queries' folder found. Please create a 'queries' folder with .sql files.")
+
+        # Set minimum users to number of defined queries if using defined queries
+        min_users = len(defined_queries) if use_defined_queries and defined_queries else 1
+        users = st.number_input(
+            "Number of synthetic users (threads)",
+            min_value=min_users, value=min_users, step=1, key="qps_users"
+        )
+        use_cache = st.checkbox("Use cache?", value=True, key="qps_use_cache")
+        optimizer_user_guided = st.checkbox("Set optimizer_mode=user_guided", value=False, key="qps_optimizer_user_guided")
+        duration = st.number_input(
+            "Test duration (seconds)",
+            min_value=1, value=30, step=1, key="qps_duration"
+        )
+
+    # Always-visible run button
+    run_test = st.button("Run QPS Test")
+
+    if run_test:
+        # Validate reasonable thread count
+        if users > 500:
+            st.warning(f"⚠️ {users} concurrent threads is very high and may cause performance issues or connection limits. Consider starting with 50-100 threads for initial testing.")
+            if users > 1000:
+                st.error(f"🚫 {users} threads exceeds recommended limits. This may crash the application or exceed database connection limits. Maximum recommended: 500 threads.")
+                st.stop()
+        # Always set cache settings before running test
+        auth = ClientCredentials(client_id=str(FIREBOLT_ACCOUNT_ID), client_secret=str(FIREBOLT_ACCOUNT_SECRET))
+        conn = connect(auth=auth, account_name=str(FIREBOLT_ACCOUNT_NAME),
+                       database=str(FIREBOLT_DATABASE), engine_name=str(FIREBOLT_ENGINE),
+                       api_endpoint=str(FIREBOLT_ENDPOINT))
+        cur = conn.cursor()
+        if optimizer_user_guided:
+            cur.execute("SET advanced_mode=1")
+            cur.execute("SET lqp2_explain_options=no_variable_types")
+            cur.execute("SET optimizer_mode=user_guided")
+
+        if not use_cache:
+            cur.execute("SET enable_result_cache = TRUE")
+            cur.execute("SET enable_subresult_cache = FALSE")
```

Comment:
Yeah, so let's swap this.
```suggestion
            cur.execute("SET enable_result_cache = FALSE")
            cur.execute("SET enable_subresult_cache = TRUE")
```